src/layouts/CustomerInvoiceDetails.tsx

```diff
@@ -174,23 +187,33 @@ export enum CustomerInvoiceDetailsTabsOptionsEnum {
   creditNotes = 'credit-notes',
 }
 
-const mapStatus = (
-  status: InvoiceStatusTypeEnum,
-  paymentStatus?: InvoicePaymentStatusTypeEnum | undefined,
-): StatusProps => {
-  if (status === InvoiceStatusTypeEnum.Draft) {
-    return { label: 'draft', type: StatusType.outline }
-  } else if (status === InvoiceStatusTypeEnum.Voided) {
-    return { label: 'voided', type: StatusType.disabled }
-  } else if (paymentStatus === InvoicePaymentStatusTypeEnum.Pending) {
-    return { label: 'pending', type: StatusType.default }
-  } else if (paymentStatus === InvoicePaymentStatusTypeEnum.Failed) {
-    return { label: 'failed', type: StatusType.warning }
-  } else if (paymentStatus === InvoicePaymentStatusTypeEnum.Succeeded) {
-    return { label: 'succeeded', type: StatusType.success }
+const getErrorMessageFromErrorDetails = (
+  errors: AllInvoiceDetailsForCustomerInvoiceDetailsFragment['errorDetails'],
+): string | undefined => {
+  if (!errors || errors.length === 0) {
+    return undefined
   }
 
-  return { label: 'n/a', type: StatusType.default }
+  const [{ errorCode, errorDetails }] = errors
+
+  if (errorCode === ErrorCodesEnum.TaxError) {
+    if (errorDetails === LagoApiError.CurrencyCodeNotSupported) {
+      return 'Currency defined is not supported for taxes calculation. Please contact the Lago team to resolve this issue.'
+    }
+
+    if (
+      errorDetails === LagoApiError.CustomerAddressCouldNotResolve ||
+      errorDetails === LagoApiError.CustomerAddressCountryNotSupported
+    ) {
+      return 'Customer address information has issues preventing calculating taxes. Please update the information to generate the invoice.'
+    }
+
+    if (errorDetails === LagoApiError.ProductExternalIdUnknown) {
+      return 'Anrok connection items mapping has issues preventing calculating taxes. Please update the mapping to generate the invoice.'
+    }
+
+    return 'An issue with your tax provider connection occurred. Please contact the Lago team to solve this issue.'
+  }
```

Comment:
TODO: Replace with https://github.com/getlago/lago-front/blob/8e484a5488ee9e45c554672477f4ea0012f7784a/src/core/constants/form.ts#L73-L78 when it's available